### ISSUE TITLE
Support Chocolatey as a 1st Class Deployment Mechanism

### ISSUE THREAD

**Issue by RehanSaeed** [dotnet/cli#7233]
The 2.0 Preview 2 SDK is not available on chocolatey. The chocolatey package exists but is maintained by the community which means some packages are not available or take time to become available. Microsoft should support the chocolatey package as a first class deployment mechanism  (Octopus Deploy are great at doing this). I want to be able to do:

`choco install dotnet-sdk --version 1.0.4`

Chocolatey is a standard release mechanism on Windows these days. It's pre-installed on `AppVeyor` and would make obtaining the .NET Core SDK smoother.

The current mechanisms for installing the SDK are to download a non-standard [PowerShell script](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script) then execute it. While this is great, it's not discoverable (someone pointed this script out to me).

**Comment by livarcocc** [comment]
This is not something supported by the CLI team itself and we don't really have plans to add it. The right way is for the community to continue supporting it.

**Comment by RehanSaeed** [comment]
I was not sure if this was the right repo to post this. The issue is that the community on chocolatey is slow and fragmented i.e. people update a package when they need it for their day jobs.

**Comment by devlead** [comment]
Kinda surprised Microsoft don't want to own their bits, with recent influx of security patches it would've been nice if Chocolatey packages was a first class citizen part of your release pipeline. Authoriting a package isn't hard, easy to automate and would be a great service to your customers. Many use Chocolatey as part of their orchestration / devops pipeline with private feeds. Know you've got your obtain ps/sh scripts but those don't work in a locked down environment.

**Comment by dracan** [comment]
Definitely, +1 for this. Chocolatey is _very_ popular, and it's extremely frustrating when packages on there aren't part of the original project's deployment pipeline, meaning that the Chocolatey packages can't be relied upon as being up to date. As @devlead said, it's not hard to automate it - then it can just be handled automatically, and everyone who uses Chocolatey can trust that this is the latest version. Also, having Microsoft as the owner increases everyone's confidence in the package.

**Comment by tannergooding** [comment]
Cc. @KathleenDollard. At least a few people want this as a supported distribution mechanism of the sdk.

**Comment by wli3** [comment]
I remember @blowdart also has security concern

**Comment by blowdart** [comment]
To say that it's a surprise we don't want to own our own bits is a little disingenuous - you're asking us to support a distribution mechanism that specifically is not our own bits, right now we own the distribution end to end. Calling it a "standard" release mechanism is also stretching it a little :)

And as @wli3 points out I've had security concerns with chocolatey, although they've been addressed over the years. Package signing would help mollify the last of my concerns.

**Comment by devlead** [comment]
@blowdart please misunderstand me correctly :) The bits are already there with hundreds of thousands of downloads total across a few packages, IMHO it would make sense if were under Microsoft control and that they were packages and released as part of the pipeline as RPMs/DEBs/etc. are released.

**Comment by KathleenDollard** [comment]
How much of the desire for Chocolatey is because we have a poor acquisition strategy?

**Comment by devlead** [comment]
@KathleenDollard In a way yes, the current strategy doesn't provide a consistent way to automate the acquisition and maintenance of SDKs/tools/etc.
 
Chocolatey is how many setup new dev machines and Windows containers/servers, through scripts/Puppet/Chef/Ansible etc.

It provides a consistent way to deploy Windows Features, tools and applications - just like you could do with something like apt-get/yum/Homebrew/etc. on other platforms.
It was originally based on NuGet, so many similarities that you can have curated private/public feeds and the community feed curated by Chocolatey.

Packages can have dependencies and you can create meta packages with just dependencies, useful for i.e. `choco install mydevmachine` or `choco install ourwebserver` and it'll pretty much be a one-liner to provision a machine/server/container.

Most NuGet feeds supports Chocolatey packages, including MyGet - which you already use, which would make it easy to provide "nightly" builds and automate testing of those, as you already do with NuGet packages.

There's already community contributed packages available for most things, including .NET Core, but perosnally think it would be a lot better if these were provided and maintained directly by the author (Microsoft), as proper self contained (or with chocolatey dependencies) chocolatey packages produced by your CI/CD pipeline. Feels even more important with the influx of security patches lately that those get out to all channels as quickly as possible.

TL;DR it would make many devs & admins happy, and provide one more way for get going with and adtopting .NET Core

**Comment by gep13** [comment]
@blowdart @KathleenDollard @tannergooding @wli3 please feel to reach out if there are any questions regarding taking ownership of the existing .net packages on chocolatey.org, or if you have any questions/concerns about Chocolatey in general.

**Comment by dracan** [comment]
@KathleenDollard It's not really anything to do with acquisition strategy. I disagree with @blowdart when he says that calling Chocolatey a standard release mechanism is stretching it. For a _very_ large number of people (especially in the developer space), Chocolatey _is_ a standard. For any software I install on Windows, the first place I look is on Chocolatey. This provides me with a very quick and reproducible installation (just by typing `cinst packagename` in the command line). Then at any time, I can type `choco outdated` to check _all_ of my installations to make sure that they're up-to-date. And I can do `cup all` to then update them all. Being able to do this across all installed software is a massive productivity win. So it's definitely not a failure in any of your strategies - it's just that Chocolatey is insanely powerful, and is used very heavily by a lot of people. As @devlead mentioned - your SDK is _already_ on Chocolatey, and being used by a lot of people. It's just not up to date. Now _this_ is your security issue - _not_ Chocolatey itself.
